agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/telemetry/ConnectionString.java

```diff
@@ -39,9 +39,19 @@
 
   private ConnectionString() {}
 
-  public static void parseInto(String connectionString, TelemetryClient targetConfig)
+  public static void parseInto(String connectionString, TelemetryClient telemetryClient)
       throws InvalidConnectionStringException {
-    mapToConnectionConfiguration(getKeyValuePairs(connectionString), targetConfig);
+    if (Strings.isNullOrEmpty(connectionString)) {
+      telemetryClient.setInstrumentationKey(connectionString);
```

Comment:
same thing, just since it looks odd to use connection string as an ikey
```suggestion
      telemetryClient.setInstrumentationKey(null);
```